src/main/java/com/github/legoatoom/connectiblechains/compat/DataDrivenCompat.java

```diff
@@ -0,0 +1,103 @@
+package com.github.legoatoom.connectiblechains.compat;
+
+import com.github.legoatoom.connectiblechains.ConnectibleChains;
+import com.github.legoatoom.connectiblechains.chain.ChainTypesRegistry;
+import com.github.legoatoom.connectiblechains.util.Helper;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import net.fabricmc.fabric.api.resource.SimpleResourceReloadListener;
+import net.minecraft.item.Item;
+import net.minecraft.resource.Resource;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.util.registry.Registry;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * This class is unused as data driven chain types are not implemented yet.
+ * Here's why:
+ * - Registries don't seem to support removing entries, so once a dynamic type is loaded
+ * it will stay until the game is restarted.
+ * - Tags don't get synced before the client and server registries get compared
+ * so a custom packet would probably be required.
+ * I (qendolin) can think of two solutions:
+ * - Use a HashMap (I would rather not)
+ * - Recreate the registry when joining a world
+ */
```

Comment:
Maybe add a comment that if anyone has ideas that they can make a pull request or create an issue with their idea.